plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -3333,6 +3935,94 @@ def reboot_factory_reset_function(self, ap_list, reboot_or_reset):
+                elif consolidated_config.get(self.keymap["mac_address"]) is not None:
+                    temp_dict[self.keymap["mac_address"]] = consolidated_config.get(self.keymap["mac_address"])
+                ap_update_list.append(temp_dict)
+                common_config.update(consolidated_config)
+
+            common_config["bulk_update"] = True
+            common_config["ap_list"] = ap_update_list
+
+            task_response = self.update_ap_configuration(common_config)
+            self.log("Access Point update response: {0} .".format(task_response), "INFO")
+            responses = {}
+            if task_response and isinstance(task_response, dict):
+                resync_retry_count = self.payload.get("dnac_api_task_timeout")
+                resync_retry_interval = self.payload.get("dnac_task_poll_interval")
+                while resync_retry_count:
+                    task_details_response = self.get_tasks_by_id(
+                        task_response["response"]["taskId"])
+                    self.log("Status of the task: {0} .".format(self.status), "INFO")
+
+                    if task_details_response.get("endTime") is not None:
+                        if task_details_response.get("status") == "FAILURE":
+                            self.result["changed"] = True if self.result["changed"] is True else False
+                            self.status = "failed"
+                            self.msg = "Unable to get success response, hence AP config not updated"
+                            self.log(self.msg, "ERROR")
+                            self.log("Task Details: {0} .".format(self.pprint(task_details_response)), "ERROR")
+                            responses["accesspoints_updates"] = {
+                                "ap_update_config_task_details": self.get_task_details_by_id(task_response["response"]["taskId"]),
+                                "ap_config_update_status": self.msg}
+                            self.module.fail_json(msg=self.msg, response=responses)
+                        else:
+                            self.result["changed"] = True
+                            self.result["ap_update_status"] = True
+                            self.log("Task Details: {0} .".format(self.pprint(
+                                task_details_response)), "INFO")
+                            self.msg = "List of AP Configuration {0} updated Successfully".format(str(ap_output_list))
+                            self.log(self.msg, "INFO")
+                            responses["accesspoints_updates"] = {
+                                "ap_update_config_task_details": self.get_task_details_by_id(task_details_response["id"]),
+                                "ap_config_update_status": self.msg
+                            }
+                            self.result["ap_update_msg"] = self.msg
+                        break
+
```

Comment:
Before invoking a sleep function (time.sleep()), add a log message to inform users of the system's pause. This message should indicate the duration of the sleep and the reason for it, such as polling for task status or waiting for a resource to become available. This helps users understand the flow of the system and the reasons for any delays.

```
self.log("Polling task status, waiting for {} seconds before the next check...".format(resync_retry_interval), "DEBUG")
time.sleep(resync_retry_interval)
```